tests/unit/base/SGObject_unittest.cc

```diff
@@ -575,3 +575,11 @@ TEST(SGObject, constrained_parameter)
     EXPECT_EQ(obj->get<int32_t>("constrained_parameter"), 1);
 }
 
+
+TEST(SGObject, auto_parameter)
+{
+    auto obj = std::make_shared<MockObject>();
+    EXPECT_THROW(obj->get<int32_t>("auto_parameter"), ShogunException);
+    obj->put("auto_parameter", 1);
```

Comment:
class member of c++ method?

Reply:
you mean test for both? I thought the SGObject unittests were all done with put/get?

Reply:
no, what I mean is we don't want to add code with hard-coded parameter names in the tests. Instead we want to use those names from public static class members, or simply use the c++ api. So could add those names to the mock class?

Reply:
Ah I see, I'll add that